### PR TITLE
ref doc generation library

### PR DESCRIPTION
**Pull request by ckairen**

### CODE REVIEW

File: packages/compiler/lib/REFDOC.md [hunk 1]
```diff
@@ -0,0 +1,497 @@
+## `@format`
```
Comment (allenjzhang) [inline packages/compiler/lib/REFDOC.md:1]: The md file format looks fine to me. Let's generate md files for other artifact types.

File: packages/doc-gen/package.json [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+        "./dist/src/index.d.ts"
+      ]
+    }
+  },
+  "cadlMain": "dist/src/index.js",
+  "engines": {
+    "node": ">=16.0.0"
+  },
+  "scripts": {
+    "clean": "rimraf ./dist ./temp",
```
Comment (allenjzhang) [inline packages/doc-gen/package.json:36]: @timotheeguerin , do we use a `npm run gendoc` script here to invoke the doc-gen process? What's the invocation step?

File: packages/doc-gen/src/index.ts [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+  Projection,
+  IdentifierNode,
+  StringLiteral,
+  SyntaxKind,
+  Type,
+  TypeReferenceNode
+} from "@cadl-lang/compiler";
+import * as fs from 'fs';
+
+export async function generateDocs(mainfile: string) {
```
Comment (allenjzhang) [inline packages/doc-gen/src/index.ts:23]: nit: @timotheeguerin, should the entry point param be a single main file or a folder?

Comment (timotheeguerin) [reply packages/doc-gen/src/index.ts:23]: We can rename that to `main` the `compile` function of the compiler just accept a path to a file or directory.

File: packages/doc-gen/src/index.ts [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+    compilerAssert(
+      node.kind === SyntaxKind.DocText,
+      "No other doc content node kinds exist yet. Update this code appropriately when more are added."
+    );
+    docs.push(node.text);
+  }
+  return docs.join("");
+}
+
+generateDocs("C:/Git/cadl/packages/compiler/lib/main.cadl");
```
Comment (timotheeguerin) [inline packages/doc-gen/src/index.ts:145]: Replace this with taking the process.argv[2]

**Comment by timotheeguerin** [comment]
Recreated a new PR with a branch on the upstream repo so we can have the website publishing be automatic #1415

**Pull request closed without merging**